main/src/cgeo/geocaching/MainActivity.java

```diff
@@ -278,6 +278,16 @@ public void onCreate(final Bundle savedInstanceState) {
         // don't call the super implementation with the layout argument, as that would set the wrong theme
         super.onCreate(savedInstanceState);
 
+        //check database
+        final String errorMsg = DataStore.initAndCheck(false);
+        if (errorMsg != null) {
+           Dialogs.message(this, "Fatal DB error - please report",
+               "Fatal error while trying to open/create cache database. " +
+                   "c:geo cannot be used in this state.\n\n" +
+                   "Please support us in analyzing and fixing this problem by report it via Settings -> System -> Generate logfile. " +
+                   "\n\n[" + errorMsg + "]");
+        }
```

Comment:
We should make this translatable, shouldn't we?

Reply:
IMHO we shouldn't as this is only a hopefully temporary debug feature...

Reply:
OK then.

Reply:
Actually, I made strings translateable now because it is now part of a bigger "user can report problem" solution